backend/translation/views.py

```diff
@@ -1,179 +1,212 @@
+            type=openapi.TYPE_STRING,
+            required=True,
+        ),
+        openapi.Parameter(
+            "target_lang",
+            openapi.IN_QUERY,
+            description=("A string to pass the target language of the translation"),
+            type=openapi.TYPE_STRING,
+            required=True,
+        ),
+        openapi.Parameter(
+            "translation_type",
+            openapi.IN_QUERY,
+            description=("A string to pass the target language of the translation"),
+            type=openapi.TYPE_STRING,
+            required=True,
+        ),
+        openapi.Parameter(
+            "get_latest",
+            openapi.IN_QUERY,
+            description=(
+                "A string to pass whether to get the latest translation or not"
+            ),
+            type=openapi.TYPE_STRING,
+            required=True,
+        ),
+    ],
+    responses={
+        200: "Generates the translation for the given transcript_id and target_lang"
+    },
+)
+@api_view(["GET"])
+def retrieve_translation(request):
+    """
+    Endpoint to retrive a translation for a given transcript and language
+    """
+    
+    # Get the query params
+    transcript_id = request.query_params.get("transcript_id")
+    target_lang = request.query_params.get("target_lang")
+    get_latest = request.query_params.get("get_latest")
+    translation_type = request.query_params.get("translation_type")
+
+    # Ensure that the UUID is valid
+    if not validate_uuid4(transcript_id):
+        return Response(
+            {"error": "Invalid transcript_id."}, status=status.HTTP_400_BAD_REQUEST
+        )
+
+    # Convert get_latest to boolean
+    get_latest = get_latest == "true"
+
+    # Ensure that required params are present
+    if not (transcript_id and target_lang and translation_type):
+        return Response(
+            {
+                "error": "Missing required query params [transcript_id, target_lang, translation_type]."
+            },
+            status=status.HTTP_400_BAD_REQUEST,
+        )
 
-    @swagger_auto_schema(
-        method="get",
-        manual_parameters=[
-            openapi.Parameter(
-                "transcript_id",
-                openapi.IN_QUERY,
-                description=("A string to pass the transcript uuid"),
+    # Get the translation for the given transcript_id, target_lang and user_id
+    queryset = (
+        Translation.objects.filter(
+            transcript_id=transcript_id,
+            target_lang=target_lang,
+            user=request.user.id,
+            translation_type=translation_type,
+        )
+        .order_by("-updated_at")
+        .first()
+    )
+    # If no translation exists for this user, check if the latest translation can be fetched
+    if queryset is None:
+        if get_latest:
+            queryset = (
+                Translation.objects.filter(
+                    transcript_id=transcript_id, target_lang=target_lang
+                )
```

Comment:
Shall we also include the `translation_type` here in the filter, similar to what we are doing in the transcript endpoint?

Reply:
Well, that depends on the functionality. This is the `get_latest` option so if we want to make it translation_type specific then we can.

Reply:
Yepp, let's do that. `get_latest` just means that we can load the latest transcript by any user.

Reply:
Cool, I will add this.